src/azure-cli/azure/cli/command_modules/identity/_params.py

```diff
@@ -17,5 +18,5 @@ def load_arguments(self, _):
         c.argument('resource_name', arg_type=name_arg_type, id_part='name')
 
     with self.argument_context('identity create') as c:
-        c.argument('location', get_location_type(self.cli_ctx))
+        c.argument('location', get_location_type(self.cli_ctx), required=False)
```

Comment:
What will happen if the user does not pass in the `--location` parameter?

Reply:
Identity can be created in other regions instead of being consistent with the resource group.

Reply:
Why is the location consistent with the resource group even if `validator=get_default_location_from_resource_group` is not used ?

Reply:
I see. This logic is implemented in `process_msi_namespace`
```
def process_msi_namespace(cmd, namespace):
    get_default_location_from_resource_group(cmd, namespace)
    validate_tags(namespace)
```